lib/tessel/deploy.js

```diff
@@ -52,89 +55,98 @@ Tessel.prototype.deployScript = function(opts, push) {
   });
 };
 
-function stopRunningScript(t, filepath) {
-  return runCommandUntilClose(t, commands.stopRunningScript(filepath));
-}
-
-function deleteOldCode(t, filepath) {
-  return runCommandUntilClose(t, commands.deleteFolder(filepath));
-}
-
-function createNewCodeFolder(t, filepath) {
-  return runCommandUntilClose(t, commands.createFolder(filepath));
-}
-
-function runCommandUntilClose(tessel, command) {
+actions.execRemoteCommand = function(tessel, command, filepath) {
   return new Promise(function(resolve) {
-    return tessel.connection.exec(command)
-      .then(function(remoteProcess) {
-        remoteProcess.once('close', resolve);
+    return tessel.connection.exec(commands[command](filepath))
+      .then(function(remote) {
+        remote.once('close', resolve);
       });
   });
-}
+};
 
-function untarStdin(t, filepath, opts, push) {
+actions.sendBundle = function(t, filepath, opts) {
   return new Promise(function(resolve, reject) {
-    // Execute the command to untar the code tarball
+    // Execute the remote untar process command
     return t.connection.exec(commands.untarStdin(filepath))
       // Once the process starts running
-      .then(function(remoteProcess) {
-        // Concat variable
-        var errData = '';
-        // If we ever get error data
-        remoteProcess.stderr.on('data', function(d) {
-          // Add it to our concat var
-          errData = d.toString();
-        });
+      .then(function(remote) {
+        var error = '';
 
-        // Include the classic Tessel library for bundling up the code
-        var tesselClassic = global.IS_TEST_ENV ?
-          require('../../test/common/tessel-classic') :
-          require('tessel');
-
-        // Gather details about the file structure of the script being sent
-        var ret = tesselClassic.analyzeScript(process.cwd() + '/' + opts.entryPoint, {
-          verbose: opts.verbose
+        // Collect error data as received
+        remote.stderr.on('data', function(data) {
+          error += data.toString();
         });
 
-        // Tar up the code to improve transfer rates
-        tesselClassic.tarCode(ret.pushdir, {
-          node: true
-        }, function(err, bundle) {
-          if (err) {
-            return reject(err);
+        actions.tarBundle(process.cwd()).then(function(bundle) {
```

Comment:
I've confirmed that I can't deploy code from another directory than the one I'm in.